python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -665,11 +667,13 @@ def reindex(self, print_fn=lambda _: None):
         print_fn("Done.")
 
     def dispose(self):
-        self._run_storage.dispose()
-        self.run_coordinator.dispose()
-        self._run_launcher.dispose()
-        self._event_storage.dispose()
-        self._compute_log_manager.dispose()
+        if not self._disposed:
+            self._disposed = True
+            self._run_storage.dispose()
+            self.run_coordinator.dispose()
+            self._run_launcher.dispose()
+            self._event_storage.dispose()
+            self._compute_log_manager.dispose()
```

Comment:
The thing I think is more pressing to error or warn on is double `dispose`. The bad recent bug we had was from exiting a context manager scope on accident causing `dispose` but still having the instance object and then proceeding to use it and then when it actually should have been `disposed` this flag was already set and resources were not cleaned up.